src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs

```diff
@@ -0,0 +1,431 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System.ComponentModel;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class EventLogWriteEntryTests
+    {
+        private const string message = "EventLogWriteEntryTestsMessage";
+
+        private readonly byte[] rawData = new byte[4] { 0, 1, 2, 3 };
+        private readonly EventInstance eventInstance = new EventInstance(0, 1);
+        private readonly string[] insertStrings = { "ExtraText", "MoreText" };
+
+        private EventLogEntry WriteLogEntry(string source, bool type = false, bool instance = false, bool category = false, bool data = false)
+        {
+            using (EventLog eventLog = new EventLog())
+            {
+                eventLog.Source = source;
+                if (instance)
+                {
+                    EventLog.WriteEvent(source, eventInstance);
+                    if (data)
+                    {
+                        eventLog.WriteEntry(message, EventLogEntryType.Warning, (int)eventInstance.InstanceId, (short)eventInstance.CategoryId, rawData);
+
+                        if (eventLog.Entries.Count == 0)
+                            return null;
+
+                        return eventLog.Entries[eventLog.Entries.Count - 1];
+                    }
+                    else if (category)
+                    {
+                        eventLog.WriteEntry(message, EventLogEntryType.Warning, (int)eventInstance.InstanceId, (short)eventInstance.CategoryId);
+
+                        if (eventLog.Entries.Count == 0)
+                            return null;
+
+                        return eventLog.Entries[eventLog.Entries.Count - 1];
+                    }
+                    else
+                    {
+                        eventLog.WriteEntry(message, EventLogEntryType.Warning, (int)eventInstance.InstanceId);
+
+                        if (eventLog.Entries.Count == 0)
+                            return null;
+
+                        return eventLog.Entries[eventLog.Entries.Count - 1];
```

Comment:
You can replace this sequence of 3 lines throughout with just `eventLog.Entries.LastOrDefault()`.

Reply:
Its nt able to cast System.Diagnostics.EventLogEntryCollection' to type 'System.Collections.Generic.IEnumerable. So not able to use this function

Reply:
Ah yes, it can cast to IEnumerable (It's an ICollection which is an IEnumerable) but it can't cast to IEnumerable<EventLogEntry> (as EventLog predates generics) and that is what Linq needs here.

You can make an extension method for IEnumerable that does the job of LastOrDefault though, if you paste this next to this class:

```c#
    internal static class EventLogEntryCollectionExtensions
    {
        internal static EventLogEntry LastOrDefault(this EventLogEntryCollection elec)
        {
            return elec.Count > 0 ? elec[elec.Count - 1] : null;
        }
    }
```

Then you can clean up this repetition.

Reply:
BTW, yo ucan do this in a follow up change if you like.